### PR TITLE
feat: add pages to vite plugin for builder

### PR DESCRIPTION
**Pull request by blaggacao**
- required by https://github.com/frappe/builder/blob/95493c56ce9e98cadca45c226c3efb51e781c36e/frontend/vite.config.js#L34

cc @surajshetty3416

### CODE REVIEW

**Comment by surajshetty3416** [comment]
@blaggacao Why is this required here 🤔

**Comment by blaggacao** [comment]
Ah, you're right. That's a better idea! Will do, shortly.

**Comment by blaggacao** [comment]
@surajshetty3416 Does this look better?

Btw, I don't know why not this diff, but I left it how it was

```diff
module.exports = function proxyOptions({
  port = 8080,
  regex = '^/(app|login|api|assets|files)',
- } = {}) {
+ }) {
```

**Comment by surajshetty3416** [comment]
> Btw, I don't know why not this diff, but I left it how it was

the `{}` in it is just a default value so that you can call `proxyOptions()` (without any parameter) and still be able to destructure the object with default values (eg. port and regex).

File: vite.js [hunk 1]
```diff
@@ -1,27 +1,30 @@
 const path = require('path')
 const fs = require('fs')
 
-module.exports = function proxyOptions({ port = 8080 } = {}) {
+module.exports = function proxyOptions({
+  port = 8080,
+  regex = '^/(app|login|api|assets|files)',
```
Comment (surajshetty3416) [inline vite.js:6]:
```suggestion
  source = '^/(app|login|api|assets|files)',
```
Variable name "source" sounds more apt? 
like "all these sources are redirected to a different target"

File: vite.js [hunk 2]
```diff
@@ -1,27 +1,30 @@
...
+ } = {}) {
   const config = getCommonSiteConfig()
   const webserver_port = config ? config.webserver_port : 8000
   if (!config) {
     console.log('No common_site_config.json found, using default port 8000')
   }
+  let proxy = {}
+  proxy[regex] = {
```
Comment (surajshetty3416) [inline vite.js:14]:
```suggestion
  proxy[source] = {
```

**Review by surajshetty3416 — CHANGES_REQUESTED** [review]

**Review by surajshetty3416 — APPROVED** [review]

**Pull request merged**